src/ast/visit.ts

```diff
@@ -146,9 +146,11 @@ export const visit = (ast: Program | AstNode, visitors: NodeVisitors) => {
               i++
             ) {
               const child = nodeValue[i - offset];
-              const res = visitNode(child, node, path, nodeKey, i - offset);
-              if (res?.removed) {
-                offset += 1;
+              if (isNode(child)) {
+                const res = visitNode(child, node, path, nodeKey, i - offset);
+                if (res?.removed) {
+                  offset += 1;
+                }
```

Comment:
I see that `child` here becomes `[ '// test', null ]` and then `'// test'` which it tries to dive into and causes the error.

At a glance I wonder if something is screwy with the parse tree. It's weird there's a `null` here. And I'm not sure the code should be getting here at all. What makes me think this, is this codepath is used for all visiting, and there are plenty of other cases where lines end with comments. 

Either way this is a definite improvement, thank you!